l10n_ch_qr_bill_scan/wizard/account_invoice_import.py

```
@@ -123,8 +123,11 @@ def _read_swiss_qr_code(self, bill_img):
         patch_img = cv2.imread(
             get_resource_path("l10n_ch_qr_bill_scan", "data", "QR-patch.png")
         )
-        y = locations[0][0]
-        x = locations[1][0]
+        try:
+            y = locations[0][0]
+            x = locations[1][0]
+        except Exception as e:
+            raise UserError(_("This QR-Code is not readable. Error: %s") % e)
```

Ok, it seems what you are looking for is to fix the line `120`

Where locations is empty. This test might be incomplete.

Side note, when using a `try...except` it is usually a bad thing to catch all exception types, it's better to narrow the catch to what we really are looking for. But, in this case I have some doubts we need it.

Would the test be more accurate with something like:

`if not locations or not locations[0]:`

Can you add a unit test for this case?

---

ok